spec/services/calculators/disqualification_calculator_spec.rb

```diff
@@ -19,26 +19,20 @@
     let(:under_age) { GenericYesNo::YES }
 
     describe "#expiry_date" do
-      context "with a lengthless than or equal to 2.5 years" do
+      context "with a length less than or equal to 2.5 years" do
         let(:conviction_length) { 2 }
         let(:conviction_length_type) { "years" }
 
-        it { expect(calculator.expiry_date.to_s).to eq((known_date + 30.months).to_s) }
+        it { expect(calculator.expiry_date.to_s).to eq((known_date + 24.months).to_s) }
       end
 
-      context "with a lengthgreater than 2.5 years" do
+      context "with a length greater than 2.5 years" do
         let(:conviction_length) { 3 }
         let(:conviction_length_type) { "years" }
 
         it { expect(calculator.expiry_date.to_s).to eq((known_date + 36.months).to_s) }
       end
 
-      context "without a length" do
```

Comment:
Has there been a change for convictions without a length?

Reply:
A disqualification with no length is spent on day of conviction.
Shall I put a test for that?

Reply:
I guess if that is really a thing that exists, then there should be a test for it.

Reply:
I have tried to resolve this with a test but my changes now give the result as 'no length' instead of spent for motoring disqualification on day of conviction.
 Hospital order also tested with no length and result given 'no length'  which may be correct I need to check.
 
 I think this is wrong